lib/js_routes/engine.rb

```diff
@@ -1,30 +1,69 @@
-require 'sprockets/version'
+class JsRoutesSprocketsExtension
+  def initialize(filename, &block)
+    @filename = filename
+    @source   = block.call
+  end
 
-class JsRoutes
-  SPROCKETS3 = Gem::Version.new(Sprockets::VERSION) >= Gem::Version.new('3.0.0')
-  class Engine < ::Rails::Engine
+  def render(context, empty_hash_wtf)
+    self.class.run(@filename, @source, context)
+  end
 
-    if SPROCKETS3
-      initializer 'js-routes.dependent_on_routes', after: :engines_blank_point, before: :finisher_hook do
-        Rails.application.config.assets.configure do |config|
+  def self.run(filename, source, context)
+    if context.logical_path == 'js-routes'
+      routes = Rails.root.join('config', 'routes.rb').to_s
+      context.depend_on(routes)
+    end
+    source
+  end
+
+  def self.call(input)
+    filename = input[:filename]
+    source   = input[:data]
+    context  = input[:environment].context_class.new(input)
+
+    result = run(filename, source, context)
+    context.metadata.merge(data: result)
+  end
+end
+
+
+class Engine < ::Rails::Engine
+  require 'sprockets/version'
+  v2                = Gem::Dependency.new('', ' ~> 2')
+  v3                = Gem::Dependency.new('', ' ~> 3')
+  v4                = Gem::Dependency.new('', ' >= 4')
+  sprockets_version = Gem::Version.new(Sprockets::VERSION).release
+  initializer_args  = case sprockets_version
+                        when -> (v) { v2.match?('', v) }
+                          { after: "sprockets.environment" }
+                        when -> (v) { v3.match?('', v) || v4.match?('', v) }
+                          { after: :engines_blank_point, before: :finisher_hook }
+                        else
+                          raise StandardError, "Sprockets version #{sprockets_version} is not supported"
+                      end
+
+  initializer 'js-routes.dependent_on_routes', initializer_args do
+    case sprockets_version
+      when -> (v) { v2.match?('', v) }
```

Comment:
Are you sure this syntax is supported by all versions of ruby that we do support?